app/frontend/src/store/projects.ts

```diff
@@ -15,17 +17,93 @@ export const $selectedProject = createStore<Project | null>(null);
 /**
  * Event to be called when selected project state should be updated
  */
-export const projectSelected = createEvent<Project | null>();
+export const projectSelected = createEvent<Project | null | undefined>();
 
 /**
  * Projects effects
  */
 export const getProjectsFx = createEffect(getProjects);
 export const createProjectFx = createEffect(createProject);
 
+
+/**
+ * Event to be called when task moved
+ */
+export const taskMoved = createEvent<{ taskId: string, prevStatusId?: string, newStatusId?: string, newIndex?: number }>();
 /**
  * State changes based on effects results
  */
 $projects.on(getProjectsFx.done, (_, { result }) => result.projects);
 $projects.on(createProjectFx.done, (state, { result }) => [...state, result.project]);
-$selectedProject.on(projectSelected, (_, value) => value);
\ No newline at end of file
+$selectedProject.on(projectSelected, (_, value) => value);
+
+/**
+ * Update project statuses once some task status changed
+ */
+$selectedProject.on(changeTaskStatusFx.done, (state, { result }) => {
```

Comment:
pls specify types of params

Reply:
no need, they are inferred

![image](https://user-images.githubusercontent.com/31101125/154800824-377e2d77-1564-460f-8239-853999e1db68.png)